src/embedded_flash/syscall.rs

```diff
@@ -41,16 +41,14 @@ mod memop_nr {
 }
 
 fn get_info(nr: usize, arg: usize) -> StorageResult<usize> {
-    let code = unsafe { syscalls::command(DRIVER_NUMBER, command_nr::GET_INFO, nr, arg) };
-    if code < 0 {
-        Err(StorageError::KernelError { code })
-    } else {
-        Ok(code as usize)
-    }
+    let code = syscalls::command(DRIVER_NUMBER, command_nr::GET_INFO, nr, arg);
+    code.map_err(|e| StorageError::KernelError {
+        code: e.return_code,
+    })
 }
 
 fn memop(nr: u32, arg: usize) -> StorageResult<usize> {
-    let code = unsafe { syscalls::memop(nr, arg) };
+    let code = unsafe { syscalls::raw::memop(nr, arg) };
     if code < 0 {
```

Comment:
Why not doing the same as above with `code.map_err()` pattern?

Reply:
Because `syscalls::raw::memop` returns a `usize` whereas `syscalls::command` returns a `Result`.